screen-access-advanced.sh

```diff
@@ -0,0 +1,80 @@
+#
+# Copyright (C) 2015  Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+# Red Hat Author(s): Vendula Poncova <vponcova@redhat.com>
+
+TESTTYPE="logs"
+
+. ${KSTESTDIR}/functions.sh
+
+log_pattern=\
+"anaconda: Spoke [a-zA-Z]\+ will not be displayed \
+because it has already been visited before.$"
+
+expected_spokes=\
+"DatetimeSpoke
+KeyboardSpoke
+LangsupportSpoke"
+
+validate() {
+    disksdir=$1
+    args=$(for d in ${disksdir}/disk-*img; do echo -a ${d}; done)
+
+    # Grab files out of the installed system while it still exists.
+    # Grab these files:
+    #
+    # logs from Anaconda - whole /var/log/anaconda/ directory is copied out,
+    #                      this can be used for saving specific test output
+    # anaconda.coverage
+    # RESULT file from the test
+    for item in /root/anaconda.coverage \
+                /var/log/anaconda/      \
+                /root/RESULT
+    do
+        virt-copy-out ${args} ${item} ${disksdir}
+    done
+
+    # The /root/RESULT file was saved from the VM.  Check its contents
+    # and decide whether the test finally succeeded or not.
+    result=$(cat ${disksdir}/RESULT)
+    if [[ $? != 0 ]]; then
+        status=1
+        echo '*** /root/RESULT does not exist in VM image.'
+    elif [[ "${result}" != SUCCESS* ]]; then
+        status=1
+        echo "${result}"
+    else
+        # Which spokes were skipped?
+        skipped_spokes=$(\
+          cat "${disksdir}/anaconda/anaconda.log" \
+        | grep -oe "${log_pattern}" \
+        | awk '{print $3;}' \
+        | sort \
+        )
+
+        # Compare with the expected spokes.
+        diff <( echo "${skipped_spokes}" ) <( echo "${expected_spokes}" )
+
+        # If the results are different, then fail.
+        if [[ $? != 0 ]]; then
+            status=1
+            echo "*** Visited spokes were not skipped."
+        fi
```

Comment:
Can't this validation be done in the kickstart file? I don't like to see `validate()` function overridden on the tests because then you can't just change/fix the validate function on one place.

So or so I'll change the validate to something which could be useful for tests to avoid copying of the `validate()` function code.